File: app/src/main/java/com/mnj/java_solutions/IntersectionOfArrays2LC350.java
package com.mnj.java_solutions;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class IntersectionOfArrays2 {

    void IntersectionOfArrays() {
        int[] nums1 = {1,2,2,1}, nums2 = {2,2};

        HashMap<Integer, Integer> map = new HashMap<>();
        List<Integer> result = new ArrayList<>();

        for (int num : nums1) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }

        for (int num : nums2) {
            if (map.containsKey(num)) {
                result.add(num);
                map.put(num, map.get(num) - 1);
                if ((map.get(num) == 0)) {
                    map.remove(num);
                }
            }
        }

        for (int i = 0; i < result.size(); i++) {
            System.out.println("==>>" + result.get(i));
        }
    }
}

File: app/src/main/java/com/mnj/java_solutions/IntersectionOfArrays2LC350.kt
package com.mnj.java_solutions

class IntersectionOfArrays2LC350 {
    fun IntersectionOfArrays() {
        val nums1 = intArrayOf(1, 2, 2, 1)
        val nums2 = intArrayOf(2, 2)
        val map = HashMap<Int, Int>()
        val result: MutableList<Int> = ArrayList()
        for (num in nums1) {
            map[num] = map.getOrDefault(num, 0) + 1
        }
        for (num in nums2) {
            if (map.containsKey(num)) {
                result.add(num)
                map[num] = map[num]!! - 1
                if (map[num] == 0) {
                    map.remove(num)
                }
            }
        }
        for (i in result.indices) {
            println("==>>" + result[i])
        }
    }
}
